lib/rest_6.js

```diff
@@ -31,6 +31,61 @@ function verbose(prompt, args) {
   }
 }
 
+/**
+ * This function constructes metadata that can be used to control the history and index flags
+ * @method{constructMetadata}
+ * @param{Object} options flags for history and indexing
+ * @param{String} contractName
+ * @returns{()} metadata
+*/
+function constructMetadata(options, contractName) {
+  const metadata = {};
+  if (options === {}) return metadata;
+
+  // history flag (default: off)
+  // {enableHistory: true}
+  if (options.enableHistory) {
+    metadata["history"] = contractName;
+  }
+  // {history: ["Contract1", "Contract2"]}
+  if (options.hasOwnProperty('history')) {
+    var newContracts = '';
+    options['history'].forEach(function(contract) {
+      if (contract !== contractName) {
+        var newContract = ',' + contract;
+        newContracts += newContract;
+      }
+    });
+    const currentHistory = options['history'];
+    const newHistory = currentHistory + newContracts;
+    metadata['history'] = newHistory;
+  }
+
+  // index flag (default: on)
+  // {enableIndex: false}
+  if (options.hasOwnProperty('enableIndex') && !options.enableIndex) {
+    metadata["noindex"] = contractName;
+  }
+  // {noindex: ["Contract1", "Contract2"]}
+  if (options.hasOwnProperty('noindex')) {
+    var newContracts = '';
```

Comment:
lines 71-77 can be replaced with one line:
const newContracts = options['noindex'].filter(contract => contract !== contractName).join()